### ISSUE TITLE
wiki docs for build.sh and rt.sh do not result in successful builds on gaea (at least)

### ISSUE THREAD

**Issue by rgrumbine** [ufs-community/ufs-weather-model#1232]
## Description
Provide a clear and concise description of what the bug is.
Also give a description of what behavior you expected to happen.

following documentation in wiki https://github.com/ufs-community/ufs-weather-model/wiki/Running-regression-test-using-rt.sh and https://github.com/ufs-community/ufs-weather-model/wiki/Building-model
does not result in a successful compilation of the UFS model.

### To Reproduce:
What compilers/machines are you seeing this with?
Give explicit steps to reproduce the behavior.

Running on gaea:

1.  at github sync my ufs-weather-model fork to the main ufs-community repository
2. source /lustre/f2/pdata/esrl/gsd/contrib/lua-5.1.4.9/init/init_lmod.sh
3. module use modulefiles
4. module load ufs_gaea.intel
5. cd ufs-weather-model
6. git pull (this is in default branch -- develop)
7. git submodule update --recursive
8. CMAKE_FLAGS="-DAPP=S2S -DCCPP_SUITES=FV3_GFS_2017_coupled,FV3_GFS_v16_coupled" ./build.sh > output 2>&1 &
    a) This fails -- referencing ccpp_prebuild.py not finding necessary xml file (details below in 'output')
9. cd tests
10. ./rt.sh -e
    a) This fails -- none of 16 compiles succeeds
    b) final line is rt.sh error on line 858
    c) more details below from compile001 (others are similar)

Same errors if I do a fresh clone of ufs-community/ufs-weather-model (error messages below are from the clone)

## Additional context

Wiki on rt.sh includes many references to the -f option, even though that is no longer functional.
rt.sh wiki does not mention the needed lmod sourcing for gaea, nor the modules to load (steps 2-4)

## Output

**output logs**

**from build.sh**
-- Found Python: /ncrc/home1/Robert.Grumbine/anaconda3/bin/python3.7 (found suitable version "3.7.0", minimum required is "3.6") found components: Interpreter 
Found Python: /ncrc/home1/Robert.Grumbine/anaconda3/bin/python3.7
Calling CCPP code generator (ccpp_prebuild.py) for suites --suites=FV3_GFS_2017_coupled,FV3_GFS_v16_coupled ...
CMake Error at FV3/ccpp/CMakeLists.txt:40 (message):
  An error occured while running ccpp_prebuild.py, check
  /lustre/f2/dev/ncep/Robert.Grumbine/fresh_ufs/build/FV3/ccpp/ccpp_prebuild.{out,err}

-- Configuring incomplete, errors occurred!
See also "/lustre/f2/dev/ncep/Robert.Grumbine/fresh_ufs/build/CMakeFiles/CMakeOutput.log"

**The ccpp_prebuild.err says:**
INFO: Parsing suite definition file suites/suite_FV3_GFS_2017_coupled.xml ...
CRITICAL: Suite definition file suites/suite_FV3_GFS_2017_coupled.xml not found.
ERROR: Parsing suite definition file suite_FV3_GFS_2017_coupled.xml failed.
Traceback (most recent call last):
  File "framework/scripts/ccpp_prebuild.py", line 761, in <module>
    main()
  File "framework/scripts/ccpp_prebuild.py", line 685, in main
    raise Exception('Parsing suite definition files failed.')
Exception: Parsing suite definition files failed.

**rt.sh -e**
**From compile_001.log:**
Job id 73575791
TEST 001 compile is waiting to enter the queue
TEST 001 compile is submitted 
Slurm unknown status Robert.Grumbine. Check sacct ...
73575791                  RUNNING          compile_001 
73575791.ba+              RUNNING                batch 
73575791.ex+              RUNNING               extern 
1 min. TEST 001 compile is RUNNING,  status: Robert.Grumbine jobid 73575791
Slurm unknown status -. Check sacct ...
73575791                   FAILED          compile_001 
73575791.ba+               FAILED                batch 
73575791.ex+            COMPLETED               extern 
2 min. TEST 001 compile is FAILED,  status: - jobid 73575791

Test 001 compile FAIL
gaea12:~/rgdev/fresh_ufs/tests/log_gaea.intel> 

**Err file from the build directory**
 + BUILD_DIR=/lustre/f2/scratch/Robert.Grumbine/FV3_RT/rt_33627/compile_001/build_fv3_001
+ [[ gaea.intel == cheyenne.* ]]
+ [[ gaea.intel == wcoss_dell_p3 ]]
+ BUILD_JOBS=8
+ hostname
+ set +x

Inactive Modules:
  1) alps      3) dvs             5) job     7) udreg     9) xpmem
  2) dmapp     4) gni-headers     6) rca     8) ugni

Lmod has detected the following error: These module(s) or extension(s) exist
but cannot be loaded as requested: "udreg/2.3.2-7.0.2.1_2.52__g8175d3d.ari"
   Try: "module spider udreg/2.3.2-7.0.2.1_2.52__g8175d3d.ari" to see how to
load the module(s).

gaea12:/lustre/f2/scratch/Robert.Grumbine/FV3_RT/rt_33627/compile_001> 

```
paste the code here (if a short section of log)
```

**Comment by MinsukJi-NOAA** [comment]
It seems that the account is not valid for Slurm jobs. Can you try the following in your shell before running `rt.sh`: `export ACCNR=nggps_emc`. Feel free to replace `nggps_emc` with some other account you know your job works under.

**Comment by rgrumbine** [comment]
I do have that export line, and it is what I used a year ago when I successfully ran rt.sh

I also used that account earlier today for an unrelated task. And ran in the queue 'normal' -- on c3.

For the rt.sh, and other, wiki, I think it'd be helpful to have a section early in the page regarding setting the environment. I only found ACCNR on the rt.sh page because you mentioned it here. Likewise the QUEUE.

**Comment by junwang-noaa** [comment]
@rgrumbine Can you do:
cd ufs-weather-model
module purge
module use -a $your_ufs-weather-model_path/modulefiles
module load ufs_gaea.intel

We assume users load ufs_gaea.intel after "module purge". We are not sure if there is anything special in your account that makes you not be able to load the module file.

**Comment by rgrumbine** [comment]
It appears that ufs_gaea.intel has many incorrect version numbers. I did the cd, purge, module use, module load, as suggested. That and the responses from the system are pasted below:

gaea12:~/rgdev/fresh_ufs> module purge
DefApps(10):ERROR:105: Unable to locate a modulefile for 'globus-toolkit'
DefApps(10):ERROR:105: Unable to locate a modulefile for 'darshan/3.2.1'
gaea12:~/rgdev/fresh_ufs> module use modulefiles
gaea12:~/rgdev/fresh_ufs> module load ufs_gaea.intel
hpc(45):ERROR:105: Unable to locate a modulefile for 'hpc/1.1.0'
ufs_gaea.intel(46):ERROR:105: Unable to locate a modulefile for 'hpc-intel/2021.3.0'
ufs_gaea.intel(47):ERROR:105: Unable to locate a modulefile for 'hpc-cray-mpich/7.7.11'
ufs_common(3):ERROR:105: Unable to locate a modulefile for 'jasper/2.0.25'
ufs_common(4):ERROR:105: Unable to locate a modulefile for 'zlib/1.2.11'
ufs_common(5):ERROR:105: Unable to locate a modulefile for 'libpng/1.6.37'
ufs_common(7):ERROR:105: Unable to locate a modulefile for 'hdf5/1.10.6'
ufs_common(8):ERROR:105: Unable to locate a modulefile for 'netcdf/4.7.4'
ufs_common(9):ERROR:105: Unable to locate a modulefile for 'pio/2.5.3'
ufs_common(10):ERROR:105: Unable to locate a modulefile for 'esmf/8.3.0b09'
ufs_common(11):ERROR:105: Unable to locate a modulefile for 'fms/2022.01'
ufs_common(13):ERROR:105: Unable to locate a modulefile for 'bacio/2.4.1'
ufs_common(14):ERROR:105: Unable to locate a modulefile for 'crtm/2.3.0'
ufs_common(15):ERROR:105: Unable to locate a modulefile for 'g2/3.4.5'
ufs_common(16):ERROR:105: Unable to locate a modulefile for 'g2tmpl/1.10.0'
ufs_common(17):ERROR:105: Unable to locate a modulefile for 'ip/3.3.3'
ufs_common(18):ERROR:105: Unable to locate a modulefile for 'sp/2.3.3'
ufs_common(19):ERROR:105: Unable to locate a modulefile for 'w3nco/2.4.1'
ufs_common(21):ERROR:105: Unable to locate a modulefile for 'gftl-shared/v1.3.3'
ufs_common(22):ERROR:105: Unable to locate a modulefile for 'yafyaml/v0.5.1'
ufs_common(23):ERROR:105: Unable to locate a modulefile for 'mapl/2.11.0-esmf-8.3.0b09'

**Comment by junwang-noaa** [comment]
@rgrumbine It looks to me that the issue is that you can't load hpc/1.1.0, which is under /lustre/f2/pdata/ncep_shared/hpc-stack/modulefiles/stack. Can you see files under that directory? What do you get if you do:

module use /lustre/f2/pdata/ncep_shared/hpc-stack/modulefiles/stack
module load hpc/1.1.0

You may need to contact nceplibs team if you can't access this module.

**Comment by rgrumbine** [comment]
the module use statement executes fine. And I've listed the contents of the directory. 
module load hpc/1.1.0
fails, as does
module load hpc/1.1.0.lua

gaea12:/lustre/f2/pdata/ncep_shared/hpc-stack/modulefiles> module use /lustre/f2/pdata/ncep_shared/hpc-stack/modulefiles/stack

gaea12:/lustre/f2/pdata/ncep_shared/hpc-stack/modulefiles> module load hpc/1.1.0
ModuleCmd_Load.c(244):ERROR:105: Unable to locate a modulefile for 'hpc/1.1.0'

gaea12:/lustre/f2/pdata/ncep_shared/hpc-stack/modulefiles> module load hpc/1.1.0.lua
locate_module.c(591):ERROR:106: Magic cookie '#%Module' missing in '/lustre/f2/pdata/ncep_shared/hpc-stack/modulefiles/stack/hpc/1.1.0.lua'
ModuleCmd_Load.c(244):ERROR:105: Unable to locate a modulefile for 'hpc/1.1.0.lua'

gaea12:/lustre/f2/pdata/ncep_shared/hpc-stack/modulefiles> dir stack/hpc
total 25
-rwxr-xr-x 1 Kyle.Gerheiser ncep 583 Aug 21  2021 1.1.0.lua
-rwxr-xr-x 1 Kyle.Gerheiser ncep 583 May  3 00:00 1.2.0.lua

**Comment by rgrumbine** [comment]
I should probably add that prior to the above load attempt, I had purged all modules.

**Comment by rgrumbine** [comment]
cd ufs-weather-model
module use modulefiles
module use /lustre/f2/pdata/ncep_shared/hpc-stack/modulefiles/stack
module purge
source /lustre/f2/pdata/esrl/gsd/contrib/lua-5.1.4.9/init/init_lmod.sh
module load ufs_gaea.intel

this does load hpc/1.1.0

but build.sh (same one as above) fails at the same point as before: ccpp_prebuild.py
rt.sh -e fails as before also.

**Comment by MinsukJi-NOAA** [comment]
For your build, I think some of the ccpp suite name in the wiki needs to be updated. You can try instead using one of the `COMPILE` lines from the `tests/rt.conf` file to build.

**Comment by rgrumbine** [comment]
rt.sh also fails to compile, using the COMPILE lines from tests/rt.conf, for all 16 compiles

**Comment by MinsukJi-NOAA** [comment]
Here is what I mean. After you successfully load all modules, change your step 8 above to
`CMAKE_FLAGS="-DAPP=S2SWA -DCCPP_SUITES=FV3_GFS_v16_coupled_nsstNoahmpUGWPv1,FV3_GFS_v17_coupled_p8" ./build.sh >out 2>&1 &`
The value for `CMAKE_FLAGS` comes from, in this case, line 5 of `tests/rt.conf`.

**Comment by rgrumbine** [comment]
**Still fails. To ensure a clean test, I logged out and back in, and made a script of the collected notes on things to do.**
#!/bin/sh

set -x

export ACCNR=nggps_emc

cd ~/rgdev/fresh_ufs

git pull
git submodule update --init --recursive

source /lustre/f2/pdata/esrl/gsd/contrib/lua-5.1.4.9/init/init_lmod.sh
module purge
module use modulefiles
module use /lustre/f2/pdata/ncep_shared/hpc-stack/modulefiles/stack
module load ufs_gaea.intel

CMAKE_FLAGS="-DAPP=S2SWA -DCCPP_SUITES=FV3_GFS_v16_coupled_nsstNoahmpUGWPv1,FV3_GFS_v17_coupled_p8" ./build.sh >out 2>&1 &


**Results are:**


++ /lustre/f2/pdata/esrl/gsd/contrib/lua-5.1.4.9/lmod/lmod/libexec/lmod bash load ufs_gaea.intel
Lmod has detected the following error: 
/opt/cray/pe/modulefiles/PrgEnv-intel/6.0.5: (PrgEnv-intel/6.0.5): can't read
"env(LOADEDMODULES)": no such variable 
While processing the following module(s):
    Module fullname     Module Filename
    ---------------     ---------------
    PrgEnv-intel/6.0.5  /opt/cray/pe/modulefiles/PrgEnv-intel/6.0.5
    ufs_gaea.intel      modulefiles/ufs_gaea.intel

+ eval false
++ false
+ CMAKE_FLAGS='-DAPP=S2SWA -DCCPP_SUITES=FV3_GFS_v16_coupled_nsstNoahmpUGWPv1,FV3_GFS_v17_coupled_p8'
+ ./build.sh
gaea14:~/rgdev> cd fresh_ufs/
gaea14:~/rgdev/fresh_ufs> cat out
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  CMake 3.19 or higher is required.  You are running version 3.10.2


-- Configuring incomplete, errors occurred!
See also "/lustre/f2/dev/ncep/Robert.Grumbine/fresh_ufs/build/CMakeFiles/CMakeOutput.log".
gaea14:~/rgdev/fresh_ufs>

**Comment by DeniseWorthen** [comment]
Gaea admin advises against using the module purge command:

WARNING: please DO NOT use the command module purge. This will remove all modules currently loaded by default in your environment and will lead to major errors. If you have accidentally used the command purge, log out of GAEA and log in. This will give you the default environment with the default modules loaded.

Should you happen to use module purge, try the following. It may or may not continue to work, but did at the time of this writing.

> module load PrgEnv-pgi> module load xt-mpich2> module load xtpe-interlagos> source /etc/csh.cshrc> source ~/.cshrc

https://gaeadocs.rdhpcs.noaa.gov/wiki/index.php/Modules

**Comment by DusanJovic-NOAA** [comment]
Change `set -x` at the beginning of the script to `set -eux` and rerun.

**Comment by rgrumbine** [comment]
Thank you Denise. I've commented out the module purge. It was recommended by jun above. Compile failed without purging also. But more has evolved since then than just the purge.

Dusan: done.

The build command listed does work now. Trying the rt.sh.

** my script that worked for build **
#!/bin/sh

set -eux

export ACCNR=nggps_emc

cd ~/rgdev/fresh_ufs

git pull
git submodule update --init --recursive

source /lustre/f2/pdata/esrl/gsd/contrib/lua-5.1.4.9/init/init_lmod.sh
#module purge
module list 2> pre.list
module use modulefiles
module use /lustre/f2/pdata/ncep_shared/hpc-stack/modulefiles/stack
module load ufs_gaea.intel
module list 2> post.list

CMAKE_FLAGS="-DAPP=S2SWA -DCCPP_SUITES=FV3_GFS_v16_coupled_nsstNoahmpUGWPv1,FV3_GFS_v17_coupled_p8" ./build.sh >out 2>&1 &

**Comment by rgrumbine** [comment]
rt.sh -e fails as before -- none of the 16 compiles succeeds.

**script for trying rt.sh:**

#!/bin/sh

set -eux

export ACCNR=nggps_emc

cd ~/rgdev/fresh_ufs

git pull
git submodule update --init --recursive

source /lustre/f2/pdata/esrl/gsd/contrib/lua-5.1.4.9/init/init_lmod.sh
#module purge
module list 2> pre.list
module use modulefiles
module use /lustre/f2/pdata/ncep_shared/hpc-stack/modulefiles/stack
module load ufs_gaea.intel
module list 2> post.list

cd tests
./rt.sh -e > rt.out 2> rt.err &

**Comment by DusanJovic-NOAA** [comment]
I still see some module load errors in:

```
$ tail /lustre/f2/scratch/Robert.Grumbine/FV3_RT/rt_24244/compile_001/err
  1) alps      3) dvs             5) job     7) udreg     9) xpmem
  2) dmapp     4) gni-headers     6) rca     8) ugni

Lmod has detected the following error: These module(s) or extension(s) exist
but cannot be loaded as requested: "udreg/2.3.2-7.0.2.1_2.52__g8175d3d.ari"
   Try: "module spider udreg/2.3.2-7.0.2.1_2.52__g8175d3d.ari" to see how to
load the module(s).
```
Do you run any module commands in your .bashrc?

**Comment by rgrumbine** [comment]
no module commands, not even the source of init_lmod.sh

There is also no .profile or .login file in my home directory.

**Comment by DusanJovic-NOAA** [comment]
go to `/lustre/f2/scratch/Robert.Grumbine/FV3_RT/rt_24244/compile_001` and try to submit `job_card` manually. If it fails, then make the job_card executable and just run it as a bash script to see if it works like that.

**Comment by DusanJovic-NOAA** [comment]
Another thing to try is to comment out `set +x` on line 58 in /lustre/f2/dev/ncep/Robert.Grumbine/fresh_ufs/tests/compile.sh:
```
 56 hostname
 57 
 58 set +x
 59 if [[ $MACHINE_ID == macosx.* ]] || [[ $MACHINE_ID == linux.* ]]; then
 60   source $PATHTR/modulefiles/ufs_${MACHINE_ID}
 61 else
 62   # Activate lua environment for gaea
 63   if [[ $MACHINE_ID == gaea.* ]] ; then
 64     source /lustre/f2/pdata/esrl/gsd/contrib/lua-5.1.4.9/init/init_lmod.sh
 65   fi
```
and resubmit the job_card.

**Comment by rgrumbine** [comment]
gaea12:~/scratch/FV3_RT/rt_24244/compile_001> sbatch job_card 
Submitted batch job 73579609 on cluster es

It ran in queue about 10 minutes, apparently successfully.

**Directory contents**

gaea9: ~/scratch/FV3_RT/rt_24244/compile_001> ls -lstr
total 175
 13 -rw-r--r-- 1 Robert.Grumbine ncep     582 May 25 14:25 job_card
137 -rw-r--r-- 1 Robert.Grumbine ncep 1637109 May 25 16:21 out
  9 -rw-r--r-- 1 Robert.Grumbine ncep      24 May 25 16:21 job_timestamp.txt
 17 -rw-r--r-- 1 Robert.Grumbine ncep  127047 May 25 16:21 err
  1 -rw-r--r-- 1 Robert.Grumbine ncep     174 May 25 16:21 compile_001_time.log
gaea9: ~/scratch/FV3_RT/rt_24244/compile_001>

**Comment by junwang-noaa** [comment]
@rgrumbine Here is the module path from emc.nemspara which is a clean account that does not load any module files initially.

-->echo $MODULEPATH
/usw/eslogin/modulefiles-c4:/sw/gaea-cle7/modulefiles/linux-sles15-x86_64:/opt/cray/pe/perftools/7.1.3/modulefiles:/opt/cray/ari/modulefiles:/opt/cray/pe/craype-targets/default/modulefiles:/opt/cray/pe/modulefiles:/opt/cray/modulefiles:/opt/modulefiles:/sw/common/modulefiles

Here are the modules loaded by the system:
```
  1) modules/3.2.11.4                                 14) dvs/2.12_2.2.177-7.0.2.1_13.5__g0b75e43d
  2) craype-network-aries                             15) alps/6.6.59-7.0.2.1_3.85__g872a8d62.ari
  3) eproxy/2.0.24-7.0.2.1_2.37__g8e04b33.ari         16) rca/2.2.20-7.0.2.1_2.93__g8e3fb5b.ari
  4) intel/19.0.5.281                                 17) atp/2.1.3
  5) craype/2.6.3                                     18) perftools-base/7.1.3
  6) cray-libsci/19.06.1                              19) PrgEnv-intel/6.0.5
  7) udreg/2.3.2-7.0.2.1_2.52__g8175d3d.ari           20) craype-broadwell
  8) ugni/6.0.14.0-7.0.2.1_3.77__ge78e5b0.ari         21) cray-mpich/7.7.11
  9) pmi/5.0.15                                       22) CmrsEnv
 10) dmapp/7.1.1-7.0.2.1_2.98__g38cf134.ari           23) TimeZoneEDT
 11) gni-headers/5.0.12.0-7.0.2.1_2.34__g3b1768f.ari  24) globus-toolkit/6.0.17
 12) xpmem/2.2.20-7.0.2.1_2.72__g87eb960.ari          25) darshan/3.2.1
 13) job/2.2.4-7.0.2.1_2.86__g36b56f4.ari             26) DefApps
```

**Comment by rgrumbine** [comment]
My .bashrc is now an empty file. There is no .login or .profile. After a fresh loging, I execute the above script (named prep_rt in my ~/rgdev directory) and it, yet again, fails all compiles.

The script includes, as the first thing to reference modules (cwd is ~/rgdev/fresh_ufs at this point):
source /lustre/f2/pdata/esrl/gsd/contrib/lua-5.1.4.9/init/init_lmod.sh
module list 2> pre.list

pre.list shows (more than this, but this is ?probably the relevant part):
++ /lustre/f2/pdata/esrl/gsd/contrib/lua-5.1.4.9/lmod/lmod/libexec/lmod bash list

Currently Loaded Modules:
  1) DefApps
  2) modules/3.2.11.4
  3) craype-network-aries
  4) eproxy/2.0.24-7.0.2.1_2.37__g8e04b33.ari
  5) intel/19.0.5.281
  6) craype/2.6.3
  7) cray-libsci/19.06.1
  8) udreg/2.3.2-7.0.2.1_2.52__g8175d3d.ari
  9) ugni/6.0.14.0-7.0.2.1_3.77__ge78e5b0.ari
 10) pmi/5.0.15
 11) dmapp/7.1.1-7.0.2.1_2.98__g38cf134.ari
 12) gni-headers/5.0.12.0-7.0.2.1_2.34__g3b1768f.ari
 13) xpmem/2.2.20-7.0.2.1_2.72__g87eb960.ari
 14) job/2.2.4-7.0.2.1_2.86__g36b56f4.ari
 15) dvs/2.12_2.2.177-7.0.2.1_13.5__g0b75e43d
 16) alps/6.6.59-7.0.2.1_3.85__g872a8d62.ari
 17) atp/2.1.3
 18) rca/2.2.20-7.0.2.1_2.93__g8e3fb5b.ari
 19) perftools-base/7.1.3
 20) PrgEnv-intel/6.0.5
 21) craype-broadwell
 22) cray-mpich/7.7.11
 23) CmrsEnv
 24) TimeZoneEDT
 25) globus-toolkit/6.0.17
 26) darshan/3.2.1

**Aside from the sequencing (mine has DefApps first rather than last) these seem to match. As far as I know, sequencing doesn't matter to this, as long as everything is present.

What doesn't seem to match is MODULEPATH:**

++ MODULEPATH=/opt/cray/pe/perftools/7.1.3/modulefiles:/opt/cray/ari/modulefiles:/usw/eslogin/modulefiles-c4:/sw/gaea-cle7/modulefiles/linux-sles15-x86_64:/opt/cray/pe/craype-targets/default/modulefiles:/opt/cray/pe/modulefiles:/opt/cray/modulefiles:/opt/modulefiles:/sw/common/modulefiles
++ export MODULEPATH

After some tedious checking, the lists include all the same directories, but the first 5 in each are in different order:
**Jun:**
/usw/eslogin/modulefiles-c4
/sw/gaea-cle7/modulefiles/linux-sles15-x86_64
/opt/cray/pe/perftools/7.1.3/modulefiles
/opt/cray/ari/modulefiles

**Bob:**
/opt/cray/pe/perftools/7.1.3/modulefiles
/opt/cray/ari/modulefiles
/usw/eslogin/modulefiles-c4
/sw/gaea-cle7/modulefiles/linux-sles15-x86_64


Sequencing may well matter for items in the MODULEPATH. I don't know. But it's certainly disconcerting that I'm doing absolutely nothing and get a different order.

**Comment by rgrumbine** [comment]
Throwing many print statements at rt.sh and run_compile.sh reveals something that looks bad about module-setup.sh

I have that script of my that does the suggested preparations before calling rt.sh. First thing after sourcing the lmod, as required for gaea, I listed out the modules available. That's the set of 26 discussed above. Last thing before rt.sh, I list what's available, and that the 52 which are mentioned above.

Inside rt.sh, I listed out the modules available at each step (my arbitrary counting of steps) and the 52 are present until module-setup.sh is sourced, at which point it's down to  25 and the order is quite different.
